File: src/dataset/typescript/nucleotide-count_exercism-wackerow/gemini/GeminiFlash_nucleotide-count_v4.ts
type Count = {
  A: number;
  C: number;
  T: number;
  G: number;
};

class NucleotideCount {
  static nucleotideCounts(_dna: string): Count {
    const dna = _dna.toUpperCase();
    if (this.isInvalid(dna)) {
      throw 'Invalid nucleotide in strand';
    }

    let aCount = 0;
    let cCount = 0;
    let tCount = 0;
    let gCount = 0;

    for (let i = 0; i < dna.length; i++) {
      const char = dna[i];
      switch (char) {
        case 'A':
          aCount++;
          break;
        case 'C':
          cCount++;
          break;
        case 'T':
          tCount++;
          break;
        case 'G':
          gCount++;
          break;
      }
    }

    return {
      A: aCount,
      C: cCount,
      T: tCount,
      G: gCount,
    };
  }

  private static isInvalid(_word: string): boolean {
    for (let i = 0; i < _word.length; i++) {
      const letter = _word[i];
      if (letter !== 'A' && letter !== 'C' && letter !== 'T' && letter !== 'G') {
        return true;
      }
    }
    return false;
  }
}

export default NucleotideCount;

File: src/dataset/cpp/nucleotide-count_exercism-Akshive/gemini/GeminiFlash_nucleotide_count_v4.cpp
#include "nucleotide_count.h"
#include <stdexcept>
#include <map>

using namespace std;

namespace dna
{
    dna::counter::counter(string DNA) : DNA(DNA)
    {
        for (char c : DNA)
        {
            if (c != 'A' && c != 'G' && c != 'T' && c != 'C')
            {
                throw std::invalid_argument("Invalid Argument");
            }
        }
    }

    map<char, int> dna::counter::nucleotide_counts() const
    {
        map<char, int> ans = {{'A', 0}, {'T', 0}, {'C', 0}, {'G', 0}};
        for (char c : DNA)
        {
            ans[c]++;
        }
        return ans;
    }

    int dna::counter::count(char c) const
    {
        if (c != 'A' && c != 'G' && c != 'T' && c != 'C')
        {
            throw std::invalid_argument("Invalid Argument");
        }
        int ans = 0;
        for (char nucleotide : DNA)
        {
            if (nucleotide == c)
            {
                ans++;
            }
        }
        return ans;
    }
}